lib/otx_ruby/hostname.rb

```diff
@@ -91,5 +91,29 @@ def get_http_scans(hostname)
 
       return results
     end
+
+    #
+    # Tallies total NIDS rules linked to a hostname
+    #
+    # @param hostname [String] Hostname to check for NIDS rules
+    # @param [Integer] Total number of NIDS rules
+    #
+    def nids_list(hostname)
+      uri = "/api/v1/indicators/hostname/#{hostname}/passive_dns"
+      json_data = get(uri)
+
+      ip_array = []
+      json_data['passive_dns'].each do |r|
+        ip_array << r['address']
+      end
+
+      total = 0
+      ip_array.each do |ip|
+        nids_list = get("/api/v1/indicators/IPv4/#{ip}/nids_list")
```

Comment:
You know you could do a call to the 'nids_list' function you use implemented elsewhere and then do the count from that?